src/components/ui/Swap/SwapHome.tsx

```diff
@@ -16,10 +16,25 @@ import { useSwapContext } from "./SwapContext";
 import { useConnectModal } from "@rainbow-me/rainbowkit";
 import { useTokenBalance } from "~/hooks/useTokenBalance";
 
+const useDebounce = (value: number, delay: number) => {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value);
+    }, delay);
+
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [value, delay]);
+  return debouncedValue;
+};
+
```

Comment:
We will use useDebounce in other parts of the project. Please extract it as a separate hook.